lib/googlesheets/archivetogooglesheets.go

```diff
@@ -116,6 +116,48 @@ func EnsureMonthColumnExists(month, year string, sheetsData SheetsData) (int, er
 	return chosenColumn, err
 }
 
+// This returns a row number (0-indexed) and a boolean as to whether a row needs to be inserted.
+//  It begins by converting the first column of the rows to string values.
+//  If there are no cells in that column, it just returns (0, false).
+//  If it comes to a cell that is empty or matches the checkName value, it returns the
+//    corresponding row number and false.
+//  If it comes to a cell that has a value greater (alphabetically) than the checkName, it
+//    returns the corresponding row number and true (i.e. a row needs to be inserted)
+func findRowPositionAndWhetherToInsertARow(checkName string, rows [][]interface{}) (int, bool) {
+	if len(rows) == 0 || len(rows[0]) == 0 {
+		return 0, false
+	}
+	checkName = strings.ToLower(checkName)
+
+	// first, just get the string values in the first column of the range
+	values := []string{}
+	for _, cells := range rows {
+		// This should never happen, but let's just be extra careful
+		if len(cells) == 0 {
+			break
+		}
+
+		value := strings.ToLower(fmt.Sprintf("%v", cells[0]))
+		values = append(values, value)
+	}
+
+	for i, value := range values {
+		if value == "" || value == checkName {
+			return i, false
+		}
+		if value > checkName {
+			return i, true
+		}
+	}
+
+	return len(values), false
```

Comment:
Do you need both loops? (Also, feel free to keep the `strings.ToLower()` as I'm not sure about the `value > checkName` here.)

```suggestion

	// first, just get the string values in the first column of the range
	for i, cells := range rows {
		// This should never happen, but let's just be extra careful
		if len(cells) == 0 {
			return i, false
		}

		value := fmt.Sprintf("%v", cells[0])
		if value == "" || strings.EqualFold(checkName, value) {
			return i, false
		}
		if value > checkName {
			return i, true
		}
	}

	return len(rows), false
```